pkg/client/apiutil/lazyrestmapper.go

```diff
@@ -199,26 +211,32 @@ func (m *lazyRESTMapper) addKnownGroupAndReload(groupName string, versions ...st
 
 // findAPIGroupByName returns API group by its name.
 func (m *lazyRESTMapper) findAPIGroupByName(groupName string) (metav1.APIGroup, error) {
```

Comment:
Could we rename this to add `findAPIGroupByName_locked` or similar to make sure callers always hold the lock?